cli/ops/runtime_compiler.rs

```diff
@@ -134,3 +139,107 @@ async fn op_emit(
     "stats": result_info.stats,
   }))
 }
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ImportMapValue {
+  imports: Value,
+  scopes: Option<Value>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum StringOrImportMap {
+  ImportMap(ImportMapValue),
+  String(String),
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct InfoArgsOptions {
+  #[serde(default)]
+  checksums: bool,
+  import_map: Option<StringOrImportMap>,
+  #[serde(default)]
+  paths: bool,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct InfoArgs {
+  specifier: String,
+  options: Option<InfoArgsOptions>,
+}
+
+async fn op_info(
+  state: Rc<RefCell<OpState>>,
+  args: InfoArgs,
+  _: (),
+) -> Result<ModuleGraphInfo, AnyError> {
+  deno_runtime::ops::check_unstable2(&state, "Deno.info");
+  let specifier = resolve_url_or_path(&args.specifier)?;
+  let program_state = state.borrow().borrow::<Arc<ProgramState>>().clone();
+  let mut runtime_permissions = {
+    let state = state.borrow();
+    state.borrow::<Permissions>().clone()
+  };
+
+  // we should check read permissions on the cache first, to ensure that we can
+  // expose any cache information to the runtime
+  let custom_root = std::env::var("DENO_DIR").map(String::into).ok();
+  let deno_dir = DenoDir::new(custom_root)?;
+  runtime_permissions
+    .check_read(&deno_dir.root)
+    .context("Deno.info() requires read access to the cache dir, run again with the --allow-read flag.")?;
+  let handler = Arc::new(Mutex::new(FetchHandler::new(
+    &program_state,
+    runtime_permissions.clone(),
+    runtime_permissions.clone(),
+  )?));
+  let maybe_import_map = if let Some(options) = &args.options {
+    match &options.import_map {
+      Some(StringOrImportMap::ImportMap(value)) => {
+        let cwd = std::env::current_dir()?;
+        let base_url = resolve_path(cwd.to_string_lossy().as_ref())?;
```

Comment:
To answer myself to previous question - in case of passing "object" import map it uses CWD as default URL, could you please add it to the JSDoc on the API?